File: LetsGo/Domain/LetsGo.Domain/Entities/Endereco.cs
using LetsGo.Domain.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;

namespace LetsGo.Domain.Entities
{
    public class Endereco
    {
        private IRestauranteRepository restauranteRepository;

        public int CodEndereco { get; set; }
        public Estado Estado { get; set; }
        public string Cidade { get; set; }
        public string Bairro { get; set; }
        public string Cep { get; set; }
        public string Rua { get; set; }
        public string Numero { get; set; }
        public string Complemento { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public Endereco(IRestauranteRepository restauranteRepository)
        {
            this.restauranteRepository = restauranteRepository;
        }

        public Endereco() { }

        public void InserirEndereco()
        {
            restauranteRepository.InserirEndereco(this);
        }

        public void InserirEndereco(IRestauranteRepository repository)
        {
            repository.InserirEndereco(this);
        }

        public void BuscarLatitudeLongitude(IRestauranteService restauranteService)
        {
            restauranteService.BuscarLatitudeLongitude(this);
        }

        public Endereco BuscarEnderecoPorRestaurante(int codRestaurante)
        {
           return restauranteRepository.BuscarEnderecoPorRestaurante(codRestaurante);
        }
    }
}

File: Mobile/LetsGo/lib/models/endereco.dart
class Endereco {

  int codEndereco;
  String uf;
  String cidade;
  String bairro;
  String cep;
  String rua;
  String numero;
  String complemento;
  double latitude;
  double longitude;

  Endereco({this.codEndereco, this.cidade, this.bairro, this.cep, this.rua, this.numero, this.complemento, this.uf, this.latitude, this.longitude});

  factory Endereco.fromJson(Map<String, dynamic> json) {
    return Endereco(
      codEndereco: json['codEndereco'],
      uf: json['estado']['sigla'],
      cidade: json['cidade'],
      bairro: json['bairro'],
      cep: json['cep'],
      rua: json['rua'],
      numero: json['numero'],
      complemento: json['complemento'],
      latitude: json['latitude'].toDouble(),
      longitude: json['longitude'].toDouble()
    );
  }

  String formatar() {
    return this.rua + ", " + 
           this.numero + " " + 
           (this.complemento == null ? "" : this.complemento) + " - " +
           this.bairro + " - " +
           this.cidade + "," + this.uf;
  }
}